src/components/ClassCardGrid.astro

```diff
@@ -34,109 +34,80 @@ const assignments = (await getCollection("docs", (entry) => entry.id.startsWith(
   },
 ]);
 
-// Static sorting of upcoming events
-const upcomingLectures = lectures.filter(lecture => +new Date(lecture.date) > +new Date()).sort((a, b) => +new Date(a.date) - +new Date(b.date));
-const upcomingAssignments = assignments.filter(assignment => +new Date(assignment.due) > +new Date()).sort((a, b) => +new Date(a.due) - +new Date(b.due));
-const upcomingExams = exams.filter(exam => +new Date(exam.date) > +new Date()).sort((a, b) => +new Date(a.date) - +new Date(b.date));
+// Get current date on the server at request time
+const currentDate = new Date();
```

Comment:
enabling SSR allows the application to dynamically generate content based on the latest data (like the current date). The line const currentDate = new Date(); retrieves that current date, ensuring that the rendered content reflects real-time information for each user request.